skglm/solvers/group_cd.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+from numpy.linalg import norm
+from numba import njit, jit
+
+from skglm.datafits.base import BaseMultitaskDatafit
+from skglm.penalties.base import BasePenalty
+
+from skglm.penalties.block_separable import SparseGroupL1
+from skglm.datafits.multi_task import QuadraticGroup
+
+from skglm.utils import make_correlated_data, grp_converter
+from celer import GroupLasso
+
+
+FREQ_CHECK = 10
+FREQ_ACC = 10
+
+
+def group_solver(X: np.ndarray, y: np.ndarray,
+                 datafit: BaseMultitaskDatafit, penalty: BasePenalty,
+                 max_iter: int = 1000, max_epochs: int = 100,
+                 stop_tol: float = 1e-7, p0: int = 10, use_acc: bool = True,
+                 verbose: bool = False) -> np.ndarray:
+
+    n_features: int = X.shape[1]
+    grp_ptr, grp_indices = penalty.grp_ptr, penalty.grp_indices
+    n_groups: int = len(grp_ptr) - 1
+
+    # init
+    w = np.zeros(n_features, dtype=np.float64)
+    Xw = X @ w  # this would make sense when w is provided
+    datafit.initialize(X, y)
+    residuals = np.array([], dtype=np.float64)
+
+    for k in range(1, max_iter+1):
+        ws = np.arange(n_groups)
+        is_optimal, scores = _check_is_optimal(datafit, penalty, y, X, w, Xw,
+                                               ws, stop_tol)
+
+        if is_optimal:
+            print("Outer Solver: early exit")
+            break
+
+        ws, ws_size = _select_ws(w, n_groups, penalty, scores, p0)
+
+        if verbose == 1:
+            p_obj = datafit.value(y, w, Xw) + penalty.value(w)
+            print(f"Iteration {k}: {p_obj}, support: {ws_size} / {n_groups}")
+
+        for epoch in range(1, max_epochs+1):
+            # inplace update of w
+            _cycle_group_cd(datafit, penalty, y, X,
```

Comment:
harmonize names with existing code for fully separable penalties
also here you're doing bcd not cd